src/services/ipfs-service.ts

```diff
@@ -174,4 +193,84 @@ export class IpfsService implements IIpfsService {
       })
     }
   }
+
+  async stop(): Promise<void> {
+    if (this.pubsub$) {
+      this.pubsub$.unsubscribe()
+    }
+
+    await this.ipfs.pubsub.unsubscribe(this.pubsubTopic)
+  }
+
+  private async createPubsub() {
+    if (this.respondToPubsubQueries) {
```

Comment:
if you invert the condition here you can early return and avoid an indentation layer which will help with readability.